### PR TITLE
User management + auth

### PR DESCRIPTION
**Pull request by psiemens**
Adds basic user management, as well as support for contributors and contributor types. Progress on #13, #9, and #14.

### CODE REVIEW

**Comment by psiemens** [comment]
Actually, you could probably say that all of those tasks are complete. You can add/delete/modify users through the default Django admin. Not sure if this counts, but it works for now.

**Comment by psiemens** [comment]
To better illustrate the relationships I formed: 

user_diagram

A User can be connected to a Person, but a Person doesn't have to be a User. A Person can be a Contributor, and a Contributor can be any number of ContributorTypes (photographer, author, illustrator, etc).

**Comment by srri-zz** [comment]
+10 beautiful diagram, try and resolve your merge conflicts locally though

File: dispatch/core/models.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            raise ValueError('User must have a valid email address')
+
+        if not self.is_valid_password(password):
+            raise ValueError('Password is invalid')
+
+        user = User(email=email,is_admin=is_admin, is_active=is_active, is_superuser=is_superuser)
+        user.set_password(password)
+        user.save()
+
+        return user
```
Comment (srri-zz) [inline dispatch/core/models.py:17]: Should it return user or should it just return a boolean value for success/fail in saving the user to the db?

**Comment by psiemens** [comment]
@Captainkrtek Cool, I'll try to figure that out

**Comment by srri-zz** [comment]
Otherwise looks good, see my one comment. I can also try to do the merge if you have issues. :+1:

File: dispatch/core/models.py (hunk 1 above)
Comment (psiemens) [inline dispatch/core/models.py:17]: It's supposed to return the new user instance that's created. For example, this allows you to create a new user and then associate them with other models (i.e. make them a contributor). I think it's also what Django expects as a return type as this is an override of a default function.

I understand where you're coming from, though. This method could be improved by adding a try/catch around the user.save() method, which I believe throws an exception if the save is unsuccessful.

**Pull request closed without merging**